plugins/chat/app/queries/chat/messages_query.rb

```diff
@@ -151,5 +157,32 @@ def self.query_paginated_messages(
         can_load_more_future: can_load_more_future,
       }
     end
+
+    def self.query_by_date(target_date, channel, messages)
+      past_messages =
+        messages
+          .where("created_at < ?", target_date)
```

Comment:
Would do `<=` for this and the future ones, that way we can be sure we don't randomly miss 1 message because it miraculously landed exactly on the datetime